scripts/staging-ci/staging-reset.js

```diff
@@ -30,15 +38,21 @@ async function main() {
     printLog(`Staging branch already up to date in ${CI_FILE}. Skipping.`)
   }
 
-  const doesStagingExist = await executeCommand(`git ls-remote --heads ${REPOSITORY} ${NEW_STAGING_BRANCH}`)
-  if (doesStagingExist) {
-    printLog('New staging branch already exists, deleting...')
-    await executeCommand(`git branch -d ${NEW_STAGING_BRANCH}`)
-    await executeCommand(`git push origin --delete ${NEW_STAGING_BRANCH}`)
+  let doesStagingExistLocally
+  try {
+    await executeCommand(`git show-ref --heads ${NEW_STAGING_BRANCH}`)
+    doesStagingExistLocally = true
+  } catch {
+    doesStagingExistLocally = false
   }
+  if (doesStagingExistLocally) {
+    printLog('New staging branch already exists locally, deleting...')
+    await executeCommand(`git branch -D ${NEW_STAGING_BRANCH}`)
+  }
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏what about directly deleting inside the try/catch?